src/withStyles.jsx

```diff
@@ -67,7 +70,23 @@ export function withStyles(
             }}
           />
         );
-      }
+      };
+    } else {
+      // WrappedComponent is a function, so we can avoid a little overhead by
+      // calling it directly instead of rendering it. This avoids React
+      // lifecycle overhead for the wrapped component and may not be necessary
+      // in a future version of React.
+      WithStyles.prototype.render = function render() {
+        const { themeName } = this.context;
+        flushIfNecessary();
+        return WrappedComponent({
```

Comment:
Overall, I'm wondering if a better approach would be to ship a semver-minor change where we add a new "optimized" entry point (and also expose the current one in an "unoptimized" entry point) - and ship that for awhile, before making the semver-major switch of which entry point the "main" reexports?